modules/sdk-unified-wallet/src/ecdsa-unified-wallet/ecdsaUnifiedWalletUtils.ts

```diff
@@ -0,0 +1,407 @@
+import {
+  KeychainsTriplet,
+  Ecdsa,
+  generateGPGKeyPair,
+  Keychain,
+  AddKeychainOptions,
+  KeyType,
+  ECDSAUtils,
+  ECDSAMethods,
+  ECDSAMethodTypes,
+  BackupKeyShare,
+  BitgoHeldBackupKeyShare,
+  BitGoBase,
+  BitgoGPGPublicKey,
+  getBitgoGpgPubKey,
+  BackupProvider,
+} from '@bitgo/sdk-core';
+import { SerializedKeyPair, Key, readKey } from 'openpgp';
+import * as openpgp from 'openpgp';
+import assert from 'assert';
+import { UnifiedWalletKeychains } from '../unifiedWalletKeychains';
+
+const encryptNShare = ECDSAMethods.encryptNShare;
+
+export class EcdsaUnifiedWalletUtils {
+  private readonly bitgo: BitGoBase;
+  private readonly _keychains: UnifiedWalletKeychains;
+  // We do not have full support for 3-party verification (w/ external source) of key shares and signature shares. There is no 3rd party key service support with this release.
+  private bitgoPublicGpgKey: openpgp.Key | undefined = undefined;
+
+  constructor(keychains: UnifiedWalletKeychains) {
+    this._keychains = keychains;
+    this.bitgo = this._keychains.getBitgo();
+    this.setBitgoGpgPubKey(this.bitgo);
+  }
+
+  private url(suffix: string): string {
+    return this._keychains.getCoin().url(suffix);
+  }
+
+  private async setBitgoGpgPubKey(bitgo) {
+    this.bitgoPublicGpgKey = await getBitgoGpgPubKey(bitgo);
+  }
+
+  async createKeychains(params: {
```

Comment:
on second thought this class looks like almost exactly like `ecdsa` 
Two question
1: what would be the major difference between ecdsa-unified-wallet and ecdsa
2. would ecdsa-unified-wallet support for example vss, uSig and etc just like ecdsa

Reply:
Also, would in future ecdsa-unified-wallet be supported for cutodial hot and trust just like Nike in ecdsa now.

So we would have backupProvider as `TrustAsKRS` then this class needs exact same modification from `ecdsa` which would be a problem to keep track the difference here and there.

Reply:
Yes it would support those things. We wanted to separate key generation from wallet creation. It looks like the wallet argument is optional for the `EcdsaUtils` class. so I was able to replace it.